pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -108,10 +108,9 @@ def setup_precision_plugin(self) -> None:
 
     def _move_optimizer_state(self, device: Optional[torch.device] = None) -> None:
         """Moves the state of the optimizers to the GPU if needed."""
-        device = device or self.root_device
         for opt in self.optimizers:
             for p, v in opt.state.items():
-                opt.state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, device)
+                opt.state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, device or self.root_device)
```

Comment:
`device = device or self.root_device` in the beginning of the method will cause issue for TPUSpawning, as the self.root_device called outside the spawning process. So the  `device = device or self.root_device` has to be in `apply_to_collection`

Reply:
Could you leave this as a comment in the code? Otherwise it'll be natural to refactor this to be at the start of the method

Reply:
Yes I'm pretty sure once we wrap Trainer's run method completely with the spawn logic #10059 (step 5), we can call root device here without probelm, and as you say, an opportunity to simplify this method.